packages/data/src/types.ts

```diff
@@ -1,40 +1,66 @@
-export type WPDataFunctionOrGeneratorArray = Array< Function | Generator >;
-export type WPDataFunctionArray = Array< Function >;
+type MapOf< T > = { [ name: string ]: T };
 
-export interface WPDataAttachedStore {
-	getSelectors: () => WPDataFunctionArray;
-	getActions: () => WPDataFunctionArray;
+export type ActionCreator = Function | Generator;
+export type Resolver = Function | Generator;
+export type Selector = Function;
+
+export type AnyConfig = StoreConfig< any, any >;
+
+export interface StoreInstance< Config extends AnyConfig > {
+	getSelectors: () => SelectorsOf< Config >;
+	getActions: () => ActionCreatorsOf< Config >;
```

Comment:
There's also the `getResolveSelectors` method that returns the promisifed selectors where one can await resolution. It's also part of the public stable API of `StoreInstance`.

All other fields are legacy.

Reply:
Thanks for clarifying. I wasn't sure if this belonged or if we wanted to expose it. All the calls I could find in practice only call this indirectly through `resolveSelect()` which seemed like it was maybe intentional that way.

Any thoughts on that?

Reply:
Also noted that we don't seem to expose this on generic stores or mention that it exists there. If we need something more specific for Redux stores maybe it'll be worth creating a `ReduxStoreInstance`